### PR TITLE
Add username validation

### PR DESCRIPTION
**Pull request by jyx11011**

### CODE REVIEW

File: src/main/webapp/WEB-INF/jsp/user-page.jsp [hunk 1]
```diff
@@ -57,7 +57,10 @@
             <label for="username-input" class="col-sm-2 col-form-label">Username</label>
             <div class="col-sm-10">
               <input type="text" class="form-control" name="username" id="username-input" placeholder="Enter a username"
-                required>
+                required pattern="^[a-zA-Z0-9_@\.]*$" maxlength="16">
```
Comment (limli) [inline src/main/webapp/WEB-INF/jsp/user-page.jsp:60]: are you going to make another PR for the backend side validation?

Comment (jyx11011) [reply src/main/webapp/WEB-INF/jsp/user-page.jsp:60]: ok will add to this PR

**Review by limli — DISMISSED** [review]

File: src/main/java/com/google/step/servlets/UserServlet.java [hunk 2]
```diff
@@ -151,4 +158,13 @@ private void updateTagsFollowedBy(long userId, String tagsString) {
 
     followManager.updateFollowedTagIds(userId, tagIds);
   }
+
+  private boolean validateUserName(String username) {
+    if (username.isEmpty()) {
+      return false;
+    }
+    Pattern p = Pattern.compile(USERNAME_PATTERN);
+    Matcher m = p.matcher(username);
```
Comment (SuyashLakhotia) [inline src/main/java/com/google/step/servlets/UserServlet.java]: Generally a bad idea to name variables this way. Use `pattern` & `matcher` instead.

**Review by SuyashLakhotia — CHANGES_REQUESTED** [review]

**Review by SuyashLakhotia — APPROVED** [review]
